entrypoint.sh

```diff
@@ -25,9 +26,30 @@ if [ -n "$MYSQL_PORT_3306_TCP" ]; then
 		echo >&2 "  Connecting to DB_HOST ($DB_HOST)"
 		echo >&2 '  instead of the linked mysql container.'
 	fi
-fi
+	: ${DB_USER:='root'}
+	if [ "$DB_USER" = 'root' ]; then
+		: ${DB_PASS:=$MYSQL_ENV_MYSQL_ROOT_PASSWORD}
+	fi
+	: ${DB_PORT:='3306'}
+elif [ -n "$POSTGRES_PORT_5432_TCP" ]; then
+	DB_TYPE=pgsql
+	if [ -z "$DB_HOST" ]; then
+		DB_HOST=$POSTGRES_PORT_5432_TCP_ADDR
+		DB_PORT=$POSTGRES_PORT_5432_TCP_PORT
+		: ${DB_NAME:='postgres'}
+	else
+		echo >&2 'WARNING: Both DB_HOST and POSTGRES_PORT_5432_TCP found.'
+		echo >&2 "  Connecting to DB_HOST ($DB_HOST)"
+		echo >&2 '  instead of the linked postgres container.'
+	fi
+	: ${DB_USER:='postgres'}
+	if [ "$DB_USER" = 'postgres' ]; then
+		: ${DB_PASS:=$POSTGRES_ENV_POSTGRES_PASSWORD}
+	fi
 
-: ${DB_PORT:='3306'}
+	echo "${DB_HOST}:${DB_PORT}:*:${DB_USER}:${DB_PASS}" > ~/.pgpass
```

Comment:
This can be removed, I found out later that drush does it automatically.

It's used because postgres is incapable of taking a password on the command line, completely refuses to do it. The pgpass file holds credentials that are used by the `psql` command.